nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1608,6 +1631,11 @@ def build_transformer_config(self) -> TransformerConfig:
 
         transformer_config = TransformerConfig(**transformer_config_dict)
 
+        # pass mcore customization configs directly to mcore
+        mcore_customization_config_dict = self.cfg.get('mcore_customization_config', {})
+        for key, value in mcore_customization_config_dict.items():
+            setattr(transformer_config, key, value)
```

Comment:
[Minor][Pedantic] Maybe we could use https://docs.python.org/3/library/dataclasses.html#dataclasses.replace for that? This has the benefit of extra validation for non-existent keys and `setattr` doesn't.

Or we add new fields with `setattr` here. Then the question would be if we should explicitely add them to `TransformerConfig`.